parser.js

```diff
@@ -3,698 +3,709 @@
+              pos++
+            }
+            return ["timestamp", result]
+          } else {
+            result += c
           }
-          return ["timestamp", result]
         } else {
-          result += c
+          err("Never happens.") // The joke is it might.
         }
-      } else {
-        err("Never happens.") // The joke is it might.
+        // 8
+        pos++
       }
-      // 8
-      pos++
     }
   }
-}
-
-var WebVTTSerializer = function() {
-  function serializeTree(tree) {
-    var result = ""
-    for (var i = 0; i < tree.length; i++) {
-      var node = tree[i]
-      if(node.type == "text") {
-        result += node.value
-      } else if(node.type == "object") {
-        result += "<" + node.name
-        if(node.classes) {
-          for(var y = 0; y < node.classes.length; y++) {
-            result += "." + node.classes[y]
+
+  var WebVTTSerializer = function() {
+    function serializeTree(tree) {
+      var result = ""
+      for (var i = 0; i < tree.length; i++) {
+        var node = tree[i]
+        if(node.type == "text") {
+          result += node.value
+        } else if(node.type == "object") {
+          result += "<" + node.name
+          if(node.classes) {
+            for(var y = 0; y < node.classes.length; y++) {
+              result += "." + node.classes[y]
+            }
           }
+          if(node.value) {
+            result += " " + node.value
+          }
+          result += ">"
+          if(node.children)
+            result += serializeTree(node.children)
+          result += "</" + node.name + ">"
+        } else {
+          result += "<" + node.value + ">"
         }
-        if(node.value) {
-          result += " " + node.value
-        }
-        result += ">"
-        if(node.children)
-          result += serializeTree(node.children)
-        result += "</" + node.name + ">"
-      } else {
-        result += "<" + node.value + ">"
       }
+      return result
     }
-    return result
-  }
-  function serializeCue(cue) {
-    return cue.startTime + " " + cue.endTime + "\n" + serializeTree(cue.tree.children) + "\n\n"
-  }
-  this.serialize = function(cues) {
-    var result = ""
-    for(var i=0;i<cues.length;i++) {
-      result += serializeCue(cues[i])
+    function serializeCue(cue) {
+      return cue.startTime + " " + cue.endTime + "\n" + serializeTree(cue.tree.children) + "\n\n"
     }
-    return result
+    this.serialize = function(cues) {
+      var result = ""
+      for(var i=0;i<cues.length;i++) {
+        result += serializeCue(cues[i])
+      }
+      return result
+    }
+  }
+
+  function exportify(object) {
+    object.WebVTTParser = WebVTTParser
+    object.WebVTTCueTimingsAndSettingsParser = WebVTTCueTimingsAndSettingsParser
+    object.WebVTTCueTextParser = WebVTTCueTextParser
+    object.WebVTTSerializer = WebVTTSerializer
   }
-}
+  if (typeof window !== 'undefined') exportify(window);
+  if (typeof exports !== 'undefined') exportify(exports);
```

Comment:
See the diff sans whitespace: https://github.com/annevk/webvtt/pull/10/files?w=1